### PR TITLE
fix: single bundle

### PR DESCRIPTION
**Pull request by wre232114**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/plugin_bundle/src/lib.rs [hunk 1]
```diff
@@ -48,7 +48,17 @@ impl Plugin for FarmPluginBundle {
...
-    let r = resource_pots.iter().map(|item| &**item).collect::<Vec<_>>();
+    let r = resource_pots
+      .iter()
+      .filter(|item| {
+        if matches!(context.config.output.target_env, TargetEnv::Library) {
+          true
+        } else {
+          matches!(item.resource_pot_type, ResourcePotType::Runtime)
+        }
+      })
```
Comment (fu050409) [inline crates/plugin_bundle/src/lib.rs:59]: Maybe you can use `matches!(context.config.output.target_env, TargetEnv::Library) || matches!(item.resource_pot_type, ResourcePotType::Runtime)`?

Comment (wre232114) [reply crates/plugin_bundle/src/lib.rs:59]: @shulandmimi

File: crates/plugin_script/src/lib.rs [hunk 2]
```diff
@@ -249,10 +251,19 @@ impl Plugin for FarmPluginScript {
     // set param.module.meta.module_system
     set_module_system_for_module_meta(param, context);
 
+    let is_replace_import_meta_url = matches!(context.config.output.target_env, TargetEnv::Library)
```
Comment (wre232114) [inline crates/plugin_script/src/lib.rs:254]: use `target_env.is_library()` instead

File: crates/plugin_script/src/lib.rs [hunk 3]
```diff
@@ -249,10 +251,19 @@ impl Plugin for FarmPluginScript {
...
+    if is_replace_import_meta_url {
+      let ast = &mut param.module.meta.as_script_mut().ast;
+      replace_import_meta_url(ast)
+    };
+
     // find and replace `import.meta.xxx` to `module.meta.xxx` and detect hmr_accepted
     // skip transform import.meta when targetEnv is node
-    if matches!(context.config.output.target_env, TargetEnv::Browser)
-      || matches!(context.config.output.format, ModuleFormat::CommonJs)
+    if !matches!(context.config.output.target_env, TargetEnv::Library)
```
Comment (wre232114) [inline crates/plugin_script/src/lib.rs:264]: use `target_env.is_browser()` instead

**Review by ErKeLost — APPROVED** [review]

**Pull request merged**